src/dos/dos_keyboard_layout.cpp

```diff
@@ -1442,151 +1618,47 @@ class DOS_KeyboardLayout final : public Module_base {
-	{"si",     COUNTRY::Slovenia       },
-	{"sk",     COUNTRY::Czech_Slovak   }, // Slovakia
-	{"sp",     COUNTRY::Spain          },
-	{"sq",     COUNTRY::Albania        }, // No-deadkeys
-	{"sq448",  COUNTRY::Albania        }, // Deadkeys
-	{"sr",     COUNTRY::Serbia         }, // Deadkey
-	{"su",     COUNTRY::Finland        },
-	{"sv",     COUNTRY::Sweden         },
-	{"sx",     COUNTRY::Spain          }, // International
-	{"tj",     COUNTRY::Tadjikistan    },
-	{"tm",     COUNTRY::Turkmenistan   },
-	{"tr",     COUNTRY::Turkey         }, // QWERTY
-	{"tr440",  COUNTRY::Turkey         }, // Non-standard
-	{"tt",     COUNTRY::Russia         }, // Tatarstan Standard
-	{"tt443",  COUNTRY::Russia         }, // Tatarstan Typewriter
-	{"ua",     COUNTRY::Ukraine        }, // 101-key
-	{"uk",     COUNTRY::United_Kingdom }, // Standard
-	{"uk168",  COUNTRY::United_Kingdom }, // Allternate
-	{"ur",     COUNTRY::Ukraine        }, // 101-key
-	{"ur465",  COUNTRY::Ukraine        }, // 101-key
-	{"ur1996", COUNTRY::Ukraine        }, // 101-key
-	{"ur2001", COUNTRY::Ukraine        }, // 102-key
-	{"ur2007", COUNTRY::Ukraine        }, // 102-key
-	{"us",     COUNTRY::United_States  }, // Standard
-	{"ux",     COUNTRY::United_States  }, // International
-	{"uz",     COUNTRY::Uzbekistan     },
-	{"vi",     COUNTRY::Vietnam        },
-	{"yc",     COUNTRY::Serbia         }, // Deadkey
-	{"yc450",  COUNTRY::Serbia         }, // No-deadkey
-	{"yu",     COUNTRY::Yugoslavia     },
-};
+const char *DOS_GetLoadedLayout();
 
-const char *DOS_GetLoadedLayout(void);
 void DOS_SetCountry(uint16_t countryNo);
-void DOS_KeyboardLayout_Init(Section* sec) {
-	test = new DOS_KeyboardLayout(sec);
-	sec->AddDestroyFunction(&DOS_KeyboardLayout_ShutDown,true);
-	// MAPPER_AddHandler(switch_keyboard_layout, SDL_SCANCODE_F2,
-	//                   MMOD1 | MMOD2, "sw_layout", "Switch Layout");
-
-	int countryNo = static_cast<Section_prop*>(sec)->Get_int("country");
-	if (!countryNo) {
-		const char *layout = DOS_GetLoadedLayout();
-		if (layout == NULL)
-			countryNo = COUNTRY::United_States;
-		else if (country_code_map.find(layout) != country_code_map.end())
-			countryNo = country_code_map.find(layout)->second;
-	}
-	DOS_SetCountry(countryNo);
+
+static void set_country_from_pref(const int country_pref)
+{
+	// default to the US
+	auto country_number = static_cast<uint16_t>(Country::United_States);
+
+	// if the user hasn't provided a country, get it from their layout
+	if (!country_pref)
+		country_number = static_cast<uint16_t>(
+		        lookup_country_from_code(DOS_GetLoadedLayout()));
+
+	// If the country they provided is valid, use that
+	else if (country_number_exists(country_pref))
+		country_number = static_cast<uint16_t>(country_pref);
+
+	// But if it's invalid, fallback to the default
+	else
+		LOG_ERR("LANGUAGE: The country number: %d could not be found, using '%u' instead.",
+		        country_pref,
+		        country_number);
```

Comment:
Personally, I find omitting braces for complex statements like this very error prone when adding/removing lines, and also a bit hard to read when multi-line statements are involved (e.g. the first `if`). Maybe we could only omit braces for simple one-liners, e.g. the previous code you removed was fine I think:

```cpp
if (layout == NULL)
	countryNo = COUNTRY::United_States;
else if (country_code_map.find(layout) != country_code_map.end())
	countryNo = country_code_map.find(layout)->second;
```

Also, when any kind of nesting is involved, I'd always just use braces, even for one-liners.

Reply:
Definitely agree - will try for that!